### DISCUSSION TITLE
Why solid-query doesn't allow mix & matching suspense & non-suspense queries

### DISCUSSION THREAD

**Discussion by rikutiira** [TanStack/query#5657] [Q&A]
First of all, thanks for all the great work on the TanStack ecosystem 🙏 

I recently started to build a solid-js project with solid-query but one thing is quite baffling to me. The [Important Differences between Solid Query & React Query documentation](https://tanstack.com/query/v4/docs/solid/overview) says:

```
- Suspense works for queries out of the box if you access the query data inside a <Suspense> boundary.
```

As a default behaviour, that is fine (albeit a bit unexpected) but you cannot even say `suspense: false` to disable suspense for a particular `createQuery` call (configuring the global `QueryClient` also doesn't do anything). To me this behaviour is not only unexpected but seems like an oversight and a missing feature. I think it's quite common that you may want to use suspense for certain queries but not for all. With the current behaviour, it seems to be impossible to have both types of queries. Moreso, the whole `options.suspense` flag doesn't do anything even though you can still give it and the types also have it.

Wondering if this a conscious design choice or an oversight and should something be done about it? Personally I'd expect the default behaviour to not use suspense which you can overwrite either globally (for `QueryClient`) or locally (for `createQuery`).

**Comment by mary-ext** [comment]
I've been bit by this, I would've thought setting `suspense` to `false` would allow for query data to be read without triggering Suspense.

I only want Suspense to be triggered in certain scenarios, and the fact that it doesn't let me do this is a huge deal-breaker.

**Comment by aadito123** [comment]
Made a PR to fix this #6415
